feat: add fortune telling app (#62167)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-fortune-teller-app/68de9361073cce61cc49cc1b.md

---
id: 68de9361073cce61cc49cc1b
title: Step 4
challengeType: 0
dashedName: step-4
---

# --description--

Well now in a previous lecture, you learned how that an interface is TypeScript's ability to automatically determine and assign types to variables, function returns, and expressions based on their usage and context.

In TypeScript, generics make functions, classes, and types more flexible.
Generics allow the creation of type variables, eliminating the need to explicitly define types each time.
This enables the same function to work safely with different types.

>  here a simple example of generics.

Reviewer: ```suggestion
Here is a simple example of generics:
```

 ```ts
 function Items<T>(list:T[]): T[] {
    return list
 }

Items<string | number>(["apple", 12,"orange"]) // here the argument is a union type
```

Functions can use two or more generics. T is used by convention, and additional type variables like S can be added as needed.

```ts
const SayHello = <S, T>(v1: S, v2: T): [S, T] => {
  return [v1, v2]
}

SayHello<string, string>("I'm", "alive") // ["I'm", "alive"]
SayHello<string, number>("I'm", 20)     // ["I'm", 20]
```

For this lesson, declare a function called `getElement` that takes `selector` as a parameter to select elements from the DOM. Use Generics so it can work with any element type.
# --hints--

> You should have declared an arrow function named `getElement`.

Reviewer: ```suggestion
You should have an arrow function named `getElement`.
```

```js
assert.match(code, /const\s*getElement\s*=/)
```

> The function should take a parameter named `selector` of type `string`.

Reviewer: ```suggestion
Your `getElement` function should take a parameter named `selector` of type `string`.
```
